### PR TITLE
LPS-135044 buildLang

### PR DESCRIPTION
**Pull request by kevenleone**

### CODE REVIEW

**Comment by kevenleone** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
ci:test:relevant

File: modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	private static final Log _log = LogFactoryUtil.getLog(
+		CheckAvailableExtensionMVCResourceCommand.class);
+
+	@Reference
+	private DLAppLocalService _dlAppLocalService;
+
+	@Reference
+	private DSEnvelopeManager _dsEnvelopeManager;
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java:103]: This variable `_dsEnvelopeManager` is never used.
You should remove it.

File: modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+				"invalidFileExtensions",
+				_jsonArray(
+					ParamUtil.getLongValues(resourceRequest, "fileEntryIds"))));
+	}
+
+	private JSONArray _jsonArray(long[] fileEntryIds) {
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray();
+
+		for (long fileEntryId : fileEntryIds) {
+			try {
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java:70]: I think is better remove try/catch and just throws Exception:

`private JSONArray _jsonArray(long[] fileEntryIds) throws Exception { ...}`

File: modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+				if (_log.isDebugEnabled()) {
+					_log.debug(exception, exception);
+				}
+			}
+		}
+
+		return jsonArray;
+	}
+
+	private static final Log _log = LogFactoryUtil.getLog(
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java:96]: Changing the method `_jsonArray` to throws Exception,  `_log` will be unnecessary.

File: modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+package com.liferay.digital.signature.web.internal.portlet.action;
+
+import com.liferay.digital.signature.constants.DigitalSignatureConstants;
+import com.liferay.digital.signature.constants.DigitalSignaturePortletKeys;
+import com.liferay.digital.signature.manager.DSEnvelopeManager;
+import com.liferay.document.library.kernel.service.DLAppLocalService;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.log.Log;
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java:24]: Remove unnecessary import.

File: modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.digital.signature.web.internal.portlet.action;
+
+import com.liferay.digital.signature.constants.DigitalSignatureConstants;
+import com.liferay.digital.signature.constants.DigitalSignaturePortletKeys;
+import com.liferay.digital.signature.manager.DSEnvelopeManager;
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java:19]: Remove unnecessary import.

File: modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import com.liferay.document.library.kernel.service.DLAppLocalService;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java:25]: Remove unnecessary import.

**Review by joseabelenda — CHANGES_REQUESTED** [review]
@kevenleone, please see my comments.
Good job!
Thanks!

File: modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+					jsonArray.put(
+						JSONUtil.put(
+							"fileEntryId", fileEntryId
+						).put(
+							"fileName", fileEntry.getFileName()
+						));
+				}
+			}
+			catch (Exception exception) {
+				if (_log.isDebugEnabled()) {
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java:87]: I think that `catch (Exception exception) {
				_log.error(exception, exception);
			}` would be better.

**Review by joseabelenda — CHANGES_REQUESTED** [review]
@kevenleone, please see my comments.
Great job.
Thanks!

File: modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java (hunk 5 above)
Comment (kevenleone) [reply modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java:19]: Thankss

File: modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java (hunk 1 above)
Comment (kevenleone) [reply modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java:103]: Thankss

File: modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java (hunk 7 above)
Comment (kevenleone) [reply modules/apps/digital-signature/digital-signature-web/src/main/java/com/liferay/digital/signature/web/internal/portlet/action/CheckAvailableExtensionMVCResourceCommand.java:87]: Thankss

**Pull request closed without merging**